dask/base.py

```diff
@@ -416,14 +416,32 @@ def tokenize(*args, **kwargs):
     return md5(str(tuple(map(normalize_token, args))).encode()).hexdigest()
 
 
+def dont_optimize(dsk, keys):
+    return dsk
+
+
+def optimization_function(obj):
+    if isinstance(obj, type):
+        cls = obj
+    else:
+        cls = type(obj)
+    name = cls.__name__.lower() + '_optimize' # dask.set_options(array_optimize=foo)
```

Comment:
There are lots of ways to consolidate the logic and still be explicit, e.g., a decorator:
```
@staticmethod
@defer_to_override('array_optimize')
def _optimize(dsk, keys):
    ...
```

It's ultimately up to you, though